src/main/java/com/amazon/ion/impl/macro/EExpressionArgsReader.java

```diff
@@ -0,0 +1,262 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.ion.impl.macro;
+
+import com.amazon.ion.IonType;
+import com.amazon.ion.SymbolToken;
+import com.amazon.ion.impl.bin.PresenceBitmap;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * An {@link EExpressionArgsReader} reads an E-Expression from a {@link ReaderAdapter}, constructs
+ * a list of {@link Expression}s representing the E-Expression and its arguments, and prepares a {@link MacroEvaluator}
+ * to evaluate these expressions.
+ * <p>
+ * There are two sources of expressions. The template macro definitions, and the macro arguments.
+ * The {@link MacroEvaluator} merges those.
+ * <p>
+ * The {@link Expression} model does not (yet) support lazily reading values, so for now, all macro arguments must
+ * be read eagerly.
+ */
+public abstract class EExpressionArgsReader {
```

Comment:
The implemented methods in here are moved from the binary reader's `EExpressionArgsReader`.